src/main/java/plugily/projects/murdermystery/arena/Arena.java

```diff
@@ -569,6 +571,27 @@ public void setGoldSpawnPoints(@NotNull List<Location> goldSpawnPoints) {
     this.goldSpawnPoints = goldSpawnPoints;
   }
 
+  public void toggleGoldVisuals() {
+    BukkitScheduler scheduler = plugin.getServer().getScheduler();
+    int id = this.goldVisualsRunnableId;
+    if (id != -1) {
+      scheduler.cancelTask(id);
+      this.goldVisualsRunnableId = -1;
+    } else {
+      this.goldVisualsRunnableId = scheduler.scheduleSyncRepeatingTask(plugin, () -> {
```

Comment:
I would rather this make async operation and checks while the gold item is valid (or on ground). Async operation for this not makes tps loss for the server and unecessarry timings report.
```suggestion
CompletableFuture.supplyAsync(() -> {
    while (item.isValid() && item.isOnGround) {
        Thread.sleep(4); // delay particle sending

        //gold.getWorld().spawnParticle
    });
}
```

This just an example, please try to find the best way.

Reply:
I'm very out of my depth with async operations. I don't use java normally so I don't think I'll be able to do that. Is it very important? I wouldn't think it matters because it's only enabled for short periods of time when nobody is playing.
Also there are no items on the ground to check because this is not happening during a game.